XbSymbolDatabase.c

```diff
@@ -247,6 +311,76 @@ extern uint32_t EmuLocateFunction(OOVPA *Oovpa, uint32_t lower, uint32_t upper)
     return 0;
 }
 
+void XbSymbolRegisterSymbol(OOVPATable* OovpaTable, uint32_t address, xb_symbol_register_t register_func) {
+    if (OovpaTable != (void*)0) {
+
+        OOVPA* Oovpa = OovpaTable->Oovpa;
+
+        // do we need to save the found address?
+        if (Oovpa->XRefSaveIndex != XRefNoSaveIndex) {
+            // If XRef is not found, save it then register once.
+            if (XRefDataBase[Oovpa->XRefSaveIndex] == 0) {
+                XRefDataBase[Oovpa->XRefSaveIndex] = address;
+                register_func((void*)0, OovpaTable->szFuncName, address, OovpaTable->Version);
+            }
+        } else {
+            register_func((void*)0, OovpaTable->szFuncName, address, OovpaTable->Version);
+        }
+    }
+}
+
+bool XbSymbolScan(void* xbeData, char* section_name, uint32_t lower_bound, uint32_t upper_bound, xb_symbol_register_t register_func) {
```

Comment:
Thanks for adding more detail. Do note that I'm not so much arguing about this, as instead getting a clear picture of what the API needs. Right now there's at least one thing not described well enough (it's still ambiguous) : what's the relation between Xbe address and section address?

Reply:
Section address is where XbSymbolDB library start scanning with OOVPA database along with . xbe_address and xbe_size is only use to enforce XREF range check similar method with XBE_MAX_VA, you're using.